### PR TITLE
Recently Used Feature Enhancement

### PR DESCRIPTION
**Pull request by sunnyjain**
# Description
Adds the Recently Used Feature in LRU manner. Fixes issue #788.

UI Changes.
### Fresh Update.
image

image

image

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: app/src/main/java/swati4star/createpdf/adapter/RecentListAdapter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    private List<String> mKeys;
+    private List<Map<String, String>> mValues;
+    private final View.OnClickListener mOnClickListener;
+
+    public RecentListAdapter(View.OnClickListener listener) {
+        this.mOnClickListener = listener;
+    }
+
+    public void updateList(List<String> keys, List<Map<String, String>> recentList) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/RecentListAdapter.java]: add javadoc style comments to this fucntion

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 2]
```diff
@@ -82,6 +90,17 @@
...
+    @BindView(R.id.recent_list)
+    RecyclerView recentList;
+
+    @BindView(R.id.recent_lbl)
+    View recentLabel;
+
+    @BindView(R.id.recent_list_lay)
+    ViewGroup recentLayout;
+
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: remove extra lines

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 3]
```diff
@@ -118,6 +137,34 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
...
 
+    @Override public void onViewCreated(
+        @NonNull final View view, @Nullable final Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+
+        RecentListAdapter adapter = new RecentListAdapter(this);
+
+        try {
+            LinkedHashMap<String, Map<String, String>> test = RecentUtil
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: rename `test` to something apt here

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: add comments

File: app/src/main/java/swati4star/createpdf/model/HomePageItem.java [hunk 4]
```diff
@@ -3,15 +3,18 @@
...
     private final int mTitleString;
 
     /**
      * Constructor for home page item
      * @param iconId - icon drawable id
+     * @param drawableId - drawable id of the Home Page Item
      * @param titleString - title string resource id of home page item
      */
-    public HomePageItem(int iconId, int titleString) {
+    public HomePageItem(int iconId, int drawableId, int titleString) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/model/HomePageItem.java]: Why is there need of this drawableId?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/model/HomePageItem.java]: Only one should work. I don't see point of bpth

File: app/src/main/java/swati4star/createpdf/util/RecentUtil.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class RecentUtil {
+
+    /**
+     * Preference key name.
+     * */
+    private static final String RECENT_PREF = "Recent";
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RecentUtil.java]: move this to constants class

**Comment by Swati4star** [comment]
What would be the UI for a fresh update?
I mean, when you dont have data for any recent used feature?

**Comment by sunnyjain** [comment]
> What would be the UI for a fresh update?
> I mean, when you dont have data for any recent used feature?

File: app/src/main/java/swati4star/createpdf/util/RecentUtil.java (hunk 5 above)
Comment (sunnyjain) [reply app/src/main/java/swati4star/createpdf/util/RecentUtil.java]: Alright.

**Comment by sunnyjain** [comment]
So when the user don't have any data as in no recently used features the section won't be visible.  I'll add More Screenshots too!.

File: app/src/main/java/swati4star/createpdf/model/HomePageItem.java (hunk 4 above)
Comment (sunnyjain) [reply app/src/main/java/swati4star/createpdf/model/HomePageItem.java]: the iconId is not actually holding the actual icon's Id which is used in the home page instead holds the menu Item Id.  I'm not sure but I understand that we can extract the icon's Id from the Menu Item. What do you think?

**Comment by Swati4star** [comment]
Great @sunnyjain

**Pull request merged**